base_delivery_carrier_extensions/models/stock_picking.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# © 2020 FactorLibre - Zahra Velasco <zahra.velasco@factorlibre.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    date_shipped = fields.Date('Shipment Date', readonly=True)
+    date_delivered = fields.Datetime('Delivery Date', readonly=True)
+    tracking_status = fields.Char('Tracking status', readonly=True, index=True)
+    tracking_status_history = fields.Text(
```

Comment:
Should we provide an extra model `delivery.tracking.status` that acts similarly to cost price in products (auto-saving a line when this is changed)?

Reply:
it's complex, the state lines can be rewritten by the carrier

Reply:
OK, it was just an idea. Maybe we are talking about different things... I'm talking about saving the moment when `carrier_state` changes from one state to another. We can get something similar putting to the field `track_visibility="onchange"`, but it's less accessible. Let's do at least this.